src/base/value_schema_manager.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#pragma once
+
+#include "pegasus_value_schema.h"
+#include <dsn/utility/singleton.h>
+
+namespace pegasus {
+
+class value_schema_manager : public dsn::utils::singleton<value_schema_manager>
+{
+public:
+    void register_schema(value_schema *schema);
+    /// using the raw value in rocksdb and data version stored in meta column family to get data
+    /// version
+    value_schema *get_value_schema(uint32_t meta_cf_data_version, dsn::string_view value) const;
+    value_schema *get_value_schema(uint32_t version) const;
+    value_schema *get_latest_value_schema() const;
+
+private:
+    value_schema_manager() = default;
+    friend class dsn::utils::singleton<value_schema_manager>;
+
+    std::map<data_version, value_schema *> _schemas;
```

Comment:
Use array instead of map. You need to ensure the cost is as least as possible.

Reply:
There won't be too many schemas, so I think the cost won't be too much.
And I use std::map here is to maintain the order for these schemas,  which will be useful in function `get_latest_value_schema`